src/nlp_preprocessing.py

```diff
@@ -1,21 +1,30 @@
-"""Preprocess the warc.gz files.
-Output = refined text, maybe a dictionary of strings, e.g.:
+from bs4 import BeautifulSoup
+import cld2
+import re
+from typing import Optional
 
-{
-    "website-1": [
-        "Lorem ipsum dolor sit amet"
-    ],
-    "website-2": [
-        "This", "is", "some", "other", "way", "of", "storing", "the", "data"
-    ],
-}
+WARCTYPE = "WARC-Type"
 
-Stuff that could happen here is: 
-- remove whitespace
-- remove stopwords
-- lemmetization
-- word stemming
-- etc.
-"""
 
-# import apache nlp?
+def preprocess_text(payload: str) -> Optional[str]:
+    for line in payload.splitlines():
+        if line.startswith(WARCTYPE) and line.split(": ")[1] != "response":
+            return
```

Comment:
Since you have `Optional[str]`, it is best practice to use `return None` in the case of returning None, instead of the implicit `return`. This improves readability.